### PR TITLE
Add provisioning job to nightly kyma on aks

### PR DESCRIPTION
**Pull request by jasiu001**
Related to [#2608](https://github.com/kyma-project/kyma/issues/2608)

### CODE REVIEW

**Comment by mszostok** [comment]
add also this entries:
```
- prefix: nightly-aks-github-integration-app-client-id
  key: client-id
- prefix: nightly-aks-github-integration-app-client-secret
  key: client-secret
```
in `prow/cluster/required-secrets.yaml` file.

Secrets already present in kyma prow and google storage

File: prow/scripts/cluster-integration/kyma-aks-long-lasting.sh [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+readonly STANDARIZED_NAME=$(echo "${INPUT_CLUSTER_NAME}" | tr "[:upper:]" "[:lower:]")
+readonly DNS_SUBDOMAIN="${STANDARIZED_NAME}"
+readonly REPO_OWNER="kyma-project"
+readonly REPO_NAME="kyma"
+readonly CURRENT_TIMESTAMP=$(date +%Y%m%d)
+
+export CLUSTER_NAME="${STANDARIZED_NAME}"
+export CLUSTER_SIZE="Standard_DS2_v2"
+export CLUSTER_K8S_VERSION="1.10.9"
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/kyma-aks-long-lasting.sh]: in minikube.sh we have: KUBERNETES_VERSION=1.11.5
please check how it is done for GKE clusters
please check if it is possible to specify wildcard value here, for example 1.10.X

File: prow/scripts/cluster-integration/kyma-aks-long-lasting.sh [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+}
+
+init
+azureAuthenticating
+
+DNS_DOMAIN="$(gcloud dns managed-zones describe "${CLOUDSDK_DNS_ZONE_NAME}" --project="${CLOUDSDK_CORE_PROJECT}" --format="value(dnsName)")"
+export DOMAIN="${DNS_SUBDOMAIN}.${DNS_DOMAIN%?}"
+
+addGithubDexConnector
+cleanup
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/kyma-aks-long-lasting.sh]: I would suggest to perform cleanup before adding github dex connector  (readability)

File: prow/scripts/cluster-integration/kyma-aks-long-lasting.sh [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    az group create --name "${RS_GROUP}" --location "${REGION}"
+}
+
+function installCluster() {
+    shout "Install Kubernetes on Azure"
+    date
+
+    echo "Find latest cluster version"
+    CLUSTER_VERSION=$(az aks get-versions -l "${REGION}" | jq '.orchestrators|.[]|select(.orchestratorVersion | contains("'"${CLUSTER_K8S_VERSION}"'"))' | jq -s '.' | jq -r 'sort_by(.orchestratorVersion | split(".") | map(tonumber)) | .[-1].orchestratorVersion')
+    echo "Latest available verion is: ${CLUSTER_VERSION}"
```
Comment (mszostok) [inline prow/scripts/cluster-integration/kyma-aks-long-lasting.sh]:
```suggestion
    echo "Latest available version is: ${CLUSTER_VERSION}"
```

File: prow/scripts/cluster-integration/kyma-aks-long-lasting.sh [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+fi
+
+# INIT ENVIRONMENT VARIABLES
+export TEST_INFRA_SOURCES_DIR="${KYMA_PROJECT_DIR}/test-infra"
+export TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS="${TEST_INFRA_SOURCES_DIR}/prow/scripts/cluster-integration/helpers"
+export KYMA_SOURCES_DIR="${KYMA_PROJECT_DIR}/kyma"
+export KYMA_SCRIPTS_DIR="${KYMA_SOURCES_DIR}/installation/scripts"
+export KYMA_RESOURCES_DIR="${KYMA_SOURCES_DIR}/installation/resources"
+
+export DOCKER_PUSH_PROJECT="kyma-installer"
```
Comment (mszostok) [inline prow/scripts/cluster-integration/kyma-aks-long-lasting.sh]: I cannot see usage of those variable, document if needed or remove

File: prow/scripts/cluster-integration/kyma-aks-long-lasting.sh [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+	        --set slackClientWebhookUrl="${SLACK_CLIENT_WEBHOOK_URL}" \
+	        --set slackClientChannelId="${STABILITY_SLACK_CLIENT_CHANNEL_ID}" \
+	        --set slackClientToken="${SLACK_CLIENT_TOKEN}" \
+	        --set stats.enabled="${STATS_ENABLED}" \
+	        --set stats.failingTestRegexp="${STATS_FAILING_TEST_REGEXP}" \
+	        --set stats.successfulTestRegexp="${STATS_SUCCESSFUL_TEST_REGEXP}" \
+	        --set testResultWindowTime="${TEST_RESULT_WINDOW_TIME}" \
+	        "${SC_DIR}/deploy/chart/stability-checker" \
+	        --namespace=kyma-system \
+	        --name=stability-checker
```
Comment (mszostok) [inline prow/scripts/cluster-integration/kyma-aks-long-lasting.sh]: please add --wait here, and also check other scripts and add there too if missing

File: prow/config.yaml [hunk 6]
```diff
@@ -66,6 +66,29 @@ presets:
     volumeMounts:
     - name: sa-kyma-artifacts
       mountPath: /etc/credentials/sa-kyma-artifacts
+  - labels:
+      preset-creds-aks-kyma-integration: "true"
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply prow/config.yaml]: 👍

File: prow/jobs/kyma/kyma-integration.yaml [hunk 7]
```diff
@@ -441,6 +441,67 @@ periodics:
...
+  cron: "0 4 * * 1-5" # "At 04:00 on every day-of-week from Monday through Friday"
+  labels:
+    preset-build-master: "true"
+    preset-stability-checker-slack-notifications: "true"
+    preset-nightly-aks-github-integration: "true"
+    preset-creds-aks-kyma-integration: "true"
+    <<: *gke_job_labels_template
+  decorate: true
+  extra_refs:
+    - org: kyma-project
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply prow/jobs/kyma/kyma-integration.yaml]: 👍

File: prow/jobs/kyma/kyma-integration.yaml [hunk 8]
```diff
@@ -441,6 +441,67 @@ periodics:
...
+          secretName: sa-stability-fluentd-storage-writer
+    containers:
+      - image: eu.gcr.io/kyma-project/test-infra/kyma-cluster-infra:v20190227-a4164f9
+        securityContext:
+          privileged: true
+        volumeMounts:
+          - name: sa-stability-fluentd-storage-writer
+            mountPath: /etc/credentials/sa-stability-fluentd-storage-writer
+            readOnly: true
+        command:
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply prow/jobs/kyma/kyma-integration.yaml]: 👍

File: prow/scripts/cluster-integration/kyma-aks-long-lasting.sh [hunk 9]
```diff
@@ -0,0 +1,328 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail  # Fail a pipe if any sub-command fails.
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply prow/scripts/cluster-integration/kyma-aks-long-lasting.sh]: 👍

File: prow/scripts/cluster-integration/kyma-aks-long-lasting.sh (hunk 3 above)
Comment (jasiu001) [reply prow/scripts/cluster-integration/kyma-aks-long-lasting.sh]: 👍

File: prow/scripts/cluster-integration/kyma-aks-long-lasting.sh (hunk 4 above)
Comment (jasiu001) [reply prow/scripts/cluster-integration/kyma-aks-long-lasting.sh]: 👍

File: prow/scripts/cluster-integration/kyma-aks-long-lasting.sh (hunk 5 above)
Comment (jasiu001) [reply prow/scripts/cluster-integration/kyma-aks-long-lasting.sh]: 👍

File: docs/prow/presets.md [hunk 10]
```diff
@@ -14,6 +14,8 @@ This document contains the list of all Presets available in the [`config.yaml`](
 | **preset-gc-compute-envs**         | It provides environment variables with the GCP compute zone and the GCP compute region.                                                                   |
 | **preset-sa-vm-kyma-integration**  | It injects credentials for the service account to run integration tests on virtual machines (VMs).                                                              |
 | **preset-sa-gke-kyma-integration** | It injects credentials for the service account to run integration tests on a Google Cloud Engine (GKE) cluster.                                                 |
+| **preset-creds-aks-kyma-integration** | It injects credentials to run integration tests on an Azure Kubernetes Service (AKS) cluster. |
+| **preset-nightly-aks-github-integration** | It injects github application credentials to enable github authentication via dex for AKS cluster |
```
Comment (klaudiagrz) [inline docs/prow/presets.md]:
```suggestion
| **preset-nightly-aks-github-integration** | It injects GitHub application credentials which enable GitHub authentication for an AKS cluster using Dex. |
```

**Pull request merged**